### ISSUE TITLE
[GTK4] Double click registers as one DOWN and two UP events instead of two of each

### ISSUE THREAD

**Issue by joel-majano** [eclipse-platform/eclipse.platform.swt#344]
Run Snippet 62 in GTK4 then double click in the window. You will see a lot of move events, and then one single "DOWN" event followed by two "UP" events. There should be two of each in the "DOWN, UP, DOWN, UP" sequence which can be obtained when using this snippet in GTK3.

Environment: Fedora 36, GTK 4.6.5, Wayland, OpenJDK17

**Comment by lathapatil** [comment]
> start the snipped with SWT_GTK4=1 env variable.

Thank you for the quick reply. Does this also apply to JUnit testing and SWT examples?

**Comment by akurtakov** [comment]
> > start the snipped with SWT_GTK4=1 env variable.
> 
> Thank you for the quick reply. Does this also apply to JUnit testing and SWT examples?

Yes, if the env variable is set SWT will try load GTK 4 libs and fallback to Gtk 3.x if loading 4.x fails.

**Comment by lathapatil** [comment]
I observed some test failures when SWT_GTK=1 was set for AllWidgetTests, eventually leading to a testcase that rendered Eclipse unresponsive and caused it to close.

Perhaps we should consider creating an issue to address these problems once somebody else confirm the behavior.

**Comment by akurtakov** [comment]
> I observed some test failures when SWT_GTK=1 was set for AllWidgetTests, eventually leading to a testcase that rendered Eclipse unresponsive and caused it to close.
> 
> Perhaps we should consider creating an issue to address these problems once somebody else confirm the behavior.

I am not aware of anyone else looking into this so this could mean waiting indefinetely.

**Comment by jonahgraham** [comment]
Confirmed in HEAD of SWT + Ubuntu 25.04 + GTK4 4.18.4+ds-1 

On GTK3 double click in the window gives this output:

```
DOWN: button: 1, stateMask=0x0, x=102, y=23
UP: button: 1, stateMask=0x80000, x=102, y=23 BUTTON1
DOWN: button: 1, stateMask=0x0, x=102, y=23
UP: button: 1, stateMask=0x80000, x=102, y=23 BUTTON1
```

And on GTK4:

```
DOWN: button: 1, stateMask=0x0, x=81, y=19
UP: button: 1, stateMask=0x80000, x=0, y=0 BUTTON1
UP: button: 1, stateMask=0x80000, x=0, y=0 BUTTON1
```

Note that in addition to the missing DOWN event, the UP events have incorrect x/y values.

**Comment by jonahgraham** [comment]
Since @Theo-Dann-Muirhead-Renesas is not a committer, I cannot assign this to him, so I have assigned it to me. He is looking at this now.

**Comment by laeubi** [comment]
@jonahgraham I think you can only assign people to an issue if they are either committers or participated in the discussion (e.g. added a comment).

**Comment by jonahgraham** [comment]
As we navigate this @Theo-Dann-Muirhead-Renesas can you add a comment that you are working on this and then I can try to assign it to you.

**Comment by Theo-Dann-Muirhead-Renesas** [comment]
Thanks for looking into this for me, Sure.

**Comment by jonahgraham** [comment]
Brilliant! Thanks @laeubi for the advice, the assignee is now correctly @Theo-Dann-Muirhead-Renesas

**Comment by Theo-Dann-Muirhead-Renesas** [comment]
This was being caused by the use of SWT.MouseDoubleClick instead of SWT.MouseDown in Control.java. I emailed @akurtakov and given how the downstream calls can be used by GTK3 as well as GTK4 I have only modified Control.java.
